services/storage/tests/test_dsm_soft_links.py

```diff
@@ -0,0 +1,133 @@
+    file.user_name = "test"
+    file.user_id = str(user_id)
+
+    async with postgres_engine.acquire() as conn:
+        stmt = (
+            file_meta_data.insert()
+            .values(
+                **attr.asdict(file),
+            )
+            .returning(literal_column("*"))
+        )
+        result = await conn.execute(stmt)
+        row = await result.fetchone()
+
+        # hacks defect
+        file.user_id = str(user_id)
+        file.location_id = str(file.location_id)
+        # --
+        assert file == FileMetaData(**dict(row))  # type: ignore
+
+        yield file
+
+        result = await conn.execute(
+            file_meta_data.delete().where(file_meta_data.c.file_uuid == row.file_uuid)
+        )
+
+
+async def test_create_soft_link(
+    storage: DataStorageManager, user_id: int, output_file: FileMetaData
+):
+
+    api_file_id = create_resource_uuid(
+        output_file.project_id, output_file.node_id, output_file.file_name
+    )
+    file_name = output_file.file_name
+
+    link_file: FileMetaDataEx = await storage.create_soft_link(
+        user_id, output_file.file_uuid, f"api/{api_file_id}/{file_name}"
+    )
+    assert isinstance(link_file, FileMetaDataEx)
+    assert isinstance(link_file.fmd, FileMetaData)
+
+    # copy:
+    #     - you have two different versions of the file.
+    #     - if you edit one, the other one stays the same.
+    #     - if you delete one, the other one stays there, but it may not be identical if it was edited
+    #     - twice as much disk space used (two different files)
+    # hard link:
+    #     - you have one file with two different filenames.
+    #     - If you edit one, it gets edited in all filename locations
+    #     - if you delete one, it still exists in other places
+    #     - only one file on disk
+    # soft link:
+    #     - you have one file with one filename and a pointer to that file with the other filename.
+    #     - if you edit the link its really editing the original file
+    #     - if you delete the file the link is broken
+    #     - if you remove the link the file stays in place.
+    #     - only one file on disk
+    #
+    #
+    #
+    # 6686134 -rw-rw-r--  1 crespo crespo 1371 Apr 14 14:53 TODO.md
+    # 6686594 -rw-rw-r--  2 crespo crespo    6 Mar  9  2020 VERSION
+    # 6686594 -rw-rw-r--  2 crespo crespo    6 Mar  9  2020 VERSION-hard
+    # 6686197 lrwxrwxrwx  1 crespo crespo    7 Apr 14 14:48 VERSION-link -> VERSION
+
+    assert link_file.fmd.file_uuid == f"api/{api_file_id}/{file_name}"
+    assert link_file.fmd.file_id == link_file.fmd.file_uuid
+    assert link_file.fmd.object_name == output_file.object_name
+    assert link_file.fmd.entity_tag == output_file.entity_tag
+    assert link_file.fmd.is_soft_link
+
+    # TODO: in principle we keep this ...
+    # assert output_file.created_at < link_file.fmd.created_at
+    # assert output_file.last_modified < link_file.fmd.last_modified
```

Comment:
Why is this disabled, reading the above makes me thing it should be enabled?

Reply:
Yes, is a flaw in the design that I am not sure how to go further with ... so i keep it until storage gets further refactored.

The idea is that storage tries to emulate linux symbolic/soft links interface. A soft link refers to the same content as the original but can have different name AND timestamps! Adding this characteristic might be handy but for the moment it complicates the server-api. Until we do not have more functionality in the storage this is not bringing any additional benefit.